### DISCUSSION TITLE
Configuration connection-line-options invalid

### DISCUSSION THREAD

**Discussion by zpChiu** [bcakmakoglu/vue-flow#1629] [Q&A] [answered]
```
<VueFlow
            @dragover="onDragOver"
            @dragleave="onDragLeave"
            @connectEnd="onConnectEnd"
            :nodes="nodes"
            :edges="edges"
            :nodeTypes="customNodeTypes"
            :connection-line-options="{type: 'smoothstep' }"
          >
            <Background
              :style="{
                backgroundColor: isDragOver ? '#e7f3ff' : 'transparent',
                transition: 'background-color 0.2s ease'
              }"
            />
</VueFlow>
```
I browse the documentation，“connection-line-options”is props attribute, So I use`{type: 'smoothstep' }`, the attribute is invalid.

Nodes are generated by dragging and dropping，Not passed in through the nodes attribute。

Can you provide an example in the documentation?

**Comment by bcakmakoglu** [comment]
> I browse the documentation，“connection-line-options”is props attribute, So I use{type: 'smoothstep' }, the attribute is invalid.

The prop is valid and works. You probably are confusing *edges* (lines that already exist) with *connection lines* (lines that you draw to **create an edge**).

https://github.com/user-attachments/assets/810b72d1-269e-47a9-9008-b791866c5857

> Nodes are generated by dragging and dropping，Not passed in through the nodes attribute。

No idea what this is supposed to mean, can you elaborate?

There are *plenty* of examples in the documentation already. What exactly are you looking for here?

**Reply by zpChiu** [reply]
I set `:connection-line-options="{type: ConnectionLineType.SmoothStep,}"`, So i think after the connection ends, the line is 'smoothstep'.

But the result is not like this. The `connection-line-options` don't change the shape of the line.

After the line connection is completed, it becomes 'smoothstep'，can you tell me how to do it. Thank you.
![微信图片_20240924092303](https://github.com/user-attachments/assets/e8a3b5aa-3b9a-487e-8a1d-c28d3356e2b2)

![Snipaste_2024-09-24_09-26-40](https://github.com/user-attachments/assets/58d720d7-7f43-481d-bb26-201a67a1f395)

**Reply by bcakmakoglu** [reply, marked as answer]
> I set :connection-line-options="{type: ConnectionLineType.SmoothStep,}", So i think after the connection ends, the line is 'smoothstep'.

A *connnection line* is **not** an *edge*. These are two different things.
If you want to use a smoothstep *edge* you can adjust the type of the edge in `onConnect` 

```ts
onConnect(params => {
	addEdges({ ...params, type: 'smoothstep' })
})
```

**Reply by zpChiu** [reply]
I understand, thank you very much for your answer.

**Reply by bcakmakoglu** [reply]
Fyi, there is also a `default-edge-options` prop which you can use to assign default properties to edges

For example:
```vue
<VueFlow :default-edge-options="{ type: 'smoothstep' }" ...>
```
